packages/cli/scripts/admin-codegen.mjs

```diff
@@ -0,0 +1,27 @@
+import {fileURLToPath} from 'node:url';
+import {ensureAuthenticatedAdmin} from '@shopify/cli-kit/node/session';
+import {generate} from '@graphql-codegen/cli';
+
+const {token} = await ensureAuthenticatedAdmin('hydrogen-preview');
```

Comment:
I'm not sure about using `hydrogen-preview` for this. Would this require logging in with that account?

Reply:
It does. It's what we use to generate all the other tokens, similar to `h2 link` using hydrogen-preview. Any other suggestion?